hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1612,6 +1612,21 @@ func (r *HostedClusterReconciler) reconcileIgnitionServer(ctx context.Context, c
 		// Reconcile route
 		ignitionServerRoute := ignitionserver.Route(controlPlaneNamespace.Name)
 		if _, err := createOrUpdate(ctx, r.Client, ignitionServerRoute, func() error {
+			// The route host is considered immutable, so set it only once upon creation
+			// and ignore updates.
+			if ignitionServerRoute.CreationTimestamp.IsZero() {
+				switch {
+				case hcluster.Spec.Platform.Type == hyperv1.AWSPlatform &&
+					(hcluster.Spec.Platform.AWS.EndpointAccess == hyperv1.PublicAndPrivate ||
```

Comment:
Nit and not introduced by you but the second half of this check is easier expressed as `hcluster.Spec.Platform.AWS.EndpointAccess != hyperv1.Public`

Reply:
Agree, gonna go ahead and merge to get CI back to stable, if somebody else is in the area next time it would be a quick refactor